code/game/machinery/recharger.dm

```diff
@@ -66,10 +66,14 @@
 			to_chat(user, SPAN_WARNING("[src] blinks red as you try to insert the item!"))
 			return
 
+		if (istype(I, /obj/item/weapon/gun/energy))
+			var/obj/item/weapon/gun/energy/W = I
+			if (W.disposable)
+				to_chat(user, SPAN_NOTICE("Your gun is disposable it cannot be charged."))
```

Comment:
Change this to "Your gun is disposable, it cannot be charged." or "Your guns battery is disposable it and so it cannot be 
be recharged."

Reply:
roger that
will do ASAP